### ISSUE TITLE
Help

### ISSUE THREAD

**Issue by lskuared** [vlachoudis/bCNC#1306]
Hello. I am new to CNC and new to Pi. Bad combo to be new at. I am a DIY'er/hobbyist, not a coder or engineer. I started with 3d printing, very intuitive. Trying to apply some of that experience to the CNC world. Not going great.

My equipment;
Pi 3b+ with Noobs
Pi 7 inch touchscreen
Arduino with CNC shield
Homemade CNC machine (3d printed parts combined with lots of Chinese metal)

As much as I have enjoyed additive printing,  I am limited to materials and would like to carve up some aluminum. Trying to crawl before I walk, I can't get things to play nicely.  Tried UGS but visualizer not working and known issue on Pi. I like bcnc but don't know how to manipulate it to my ends.

I have Googled everything but not everyone has the exact same scenarios. Lots of answers are 'just enter g98-tt56 and good to go'. Kill me now.

I have explored github and wiki's to no avail.

Hope you can help and in plain English if possible. 

How do I tell bcnc my build area? Bcnc opens and has an orange square in the preview window. Is that my supposed build area?

How do I tell bcnc the size of my material? Is that what the orange square is? 

Why is that when I load a file it loads to random locations on the preview window but never  centered in the orange square?  Is it supposed to? 

I open/upload a file and it is tiny. How do I scale it up to fit the build area better?

I have uploaded 3 images. Showing how 2 different files load to two different locations.  Other image is my machine with pencil.

![20191114_224312](https://user-images.githubusercontent.com/57068911/68921098-c56a6700-0734-11ea-88ec-90255126629e.jpg)
![20191114_224546](https://user-images.githubusercontent.com/57068911/68921099-c56a6700-0734-11ea-9177-b2961d6f48a9.jpg)
![20191114_224329](https://user-images.githubusercontent.com/57068911/68921100-c56a6700-0734-11ea-9fe2-fae02de382c7.jpg)

Thanks in advance, 
Frustrated (Jim)

**Comment by Harvie** [comment]
Hello, i see you've built Nikodem Bartnik's Dremel CNC :-)

> How do I tell bcnc my build area? Bcnc opens and has an orange square in the preview window. Is that my supposed build area?

Don't worry. Just click the button with orange square. It will hide the work area. You don't really need that in the beginning. If you really want to do that, you can specify it in CAM->config menu as "travel X,Y,Z" items, but as beginner you really don't need to do this.

> How do I tell bcnc the size of my material? Is that what the orange square is?

You don't really need to tell the bCNC size of your material. Except for it's thickness.
Set surface Z to 0, Set safe Z to whatewer clearance you need for rapid motion over the stock (so you don't hit anything when traveling between cuts) and set thickness to something reasonable. bCNC will not allow you to cut deeper than the material thickness. It's just sanity check. I personaly don't like it, so i set something like 500mm, so it does not bother me. You will set actual cutting depth when preparing individual cuts, so it does not really affect anything.

![image](https://user-images.githubusercontent.com/276504/69913021-bf61cf00-1432-11ea-90e2-14a1c87c0645.png)

> Why is that when I load a file it loads to random locations on the preview window but never centered in the orange square? Is it supposed to?

As i said, you're better of disabling the orange square before you learn the workflow. It's not important. What is important is so called "origin" or "zero" point. This is where your drawing has 0,0 coordinates. You than jog your machine to that XY location and jog Z down so dat the mill just touches the top of your material. Then you click XYZ=0 button in bCNC and you are ready to go. Every toolpath in your g-code will be then referenced to that physical point.

What you choose as reference point is up to you. if you have big piece of material and want to use it as efficiently as possible, you usualy put the origin in the corner. If you have small piece of material and need to fit the part in it, you can use center of the material as origin. Sometimes your cuting needs to be relative to some feature that is already made in the milled object, so you have to take that as reference. You need to prepare your CAD design so that origin of drawing is where you need it and where you can jog the machine and zero the coordinates by touching the material.

If you didn't prepared the origin point in CAD correctly, you can modify it later in bCNC:

![image](https://user-images.githubusercontent.com/276504/69913196-de616080-1434-11ea-8d48-db3ca653a4ee.png)

When you have your origin set properly, you can jog 10mm Z up or so, so you don't hit anything and click this "scan" button, which will make your machine do XY movement around bounding box of your g-code, so you will easily check if everything fits:

![image](https://user-images.githubusercontent.com/276504/69913249-cccc8880-1435-11ea-9f53-e64177979425.png)

> I open/upload a file and it is tiny. How do I scale it up to fit the build area better?

Preferred way is to edit the design in CAD. You should also check that you have your machine/GRBL properly calibrated and set everything to metric rather than inches (unless you've been contracted with imperial disease lol).

If you are getting the true dimensions, you can obviously scale the design. In the CAD or maybe even the bCNC. There was some scaling plugin recently added, but i haven't heavily tested it yet, because i mostly prepare all my drawings in LibreCAD and use bCNC just for the CAM work and streaming to machine.
